homeassistant/components/websocket_api/automation.py

```diff
@@ -0,0 +1,149 @@
+    domains: set[str] = set()
+    device_classes: set[str] = set()
+    supported_features: list[int] = []
+
+    selector_entities = target_description.get("entity", [])
+    for selector_entity in selector_entities:
+        domains.update(selector_entity.get("domain", []))
+        device_classes.update(selector_entity.get("device_class", []))
+        supported_features.extend(selector_entity.get("supported_features", []))
+
+    return _ComponentLookupData(
+        component=component,
+        domains=domains,
+        device_classes=device_classes,
+        supported_features=supported_features,
+    )
+
+
+async def _async_get_components_for_target(
+    hass: HomeAssistant,
+    target_selector: ConfigType,
+    expand_group: bool,
+    descriptions: dict[str, dict[str, Any] | None],
+) -> set[str]:
+    """Get automation components (triggers/conditions/services) for a target.
+
+    Returns all components that can be used on any entity that are currently part of a target.
+    """
+    selector_data = target_helpers.TargetSelectorData(target_selector)
+    extracted = target_helpers.async_extract_referenced_entity_ids(
+        hass, selector_data, expand_group=expand_group
+    )
+    _LOGGER.debug("Extracted entities for lookup: %s", extracted)
+
+    # Build lookup structure: domain -> list of trigger/condition/service lookup data
+    domain_components: dict[str, list[_ComponentLookupData]] = {}
+    component_count = 0
+    for component, description in descriptions.items():
+        if description is None or CONF_TARGET not in description:
+            _LOGGER.debug("Skipping component %s without target description", component)
+            continue
+        domain = component.split(".")[0]
+        domain_components.setdefault(domain, []).append(
+            _build_component_lookup_data(component, description[CONF_TARGET])
+        )
+        component_count += 1
+
+    _LOGGER.debug("Domain components: %s", domain_components)
+
+    matched_components: set[str] = set()
+
+    entity_registry = er.async_get(hass)
+    for entity_id in extracted.referenced.union(extracted.indirectly_referenced):
+        if component_count == len(matched_components):
+            break
+
+        if (entity_entry := entity_registry.async_get(entity_id)) is None:
+            continue
+
+        for domain in (entity_entry.domain, entity_entry.platform):
+            for component_data in domain_components.get(domain, []):
+                if component_data.component in matched_components:
+                    continue
+                if component_data.matches(entity_entry):
+                    matched_components.add(component_data.component)
+
+    return matched_components
+
+
+async def async_get_triggers_for_target(
+    hass: HomeAssistant, target_selector: ConfigType, expand_group: bool
+) -> set[str]:
+    """Get triggers for a target."""
+    descriptions = await async_get_all_trigger_descriptions(hass)
```

Comment:
Remove this single use variable

Reply:
why? I don't think making the call to `_async_get_automation_components_for_target` a oneliner improves readability here.